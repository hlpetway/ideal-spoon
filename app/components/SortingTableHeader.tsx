import TableSortLabel from "@mui/material/TableSortLabel";
import TableHead from "@mui/material/TableHead";
import TableCell from "@mui/material/TableCell";
import TableRow from "@mui/material/TableRow";
import { Box } from "@mui/material";
import { visuallyHidden } from '@mui/utils';
import { Sale } from './SalesTable';
import { Order } from "../types";

interface HeadCell {
    id: keyof Sale;
    label: string;
}

const headCells: readonly HeadCell[] = [
    {
        id: "weekEnding",
        label: "WEEK ENDING",
    },
    {
        id: "retailSales",
        label: "RETAIL SALES",
    },
    {
        id: "wholesaleSales",
        label: "WHOLESALE SALES",
    },
    {
        id: "unitsSold",
        label: "UNITS SOLD",
    },
    {
        id: "retailerMargin",
        label: "RETAILER MARGIN",
    },
];

interface EnhancedTableProps {
    onRequestSort: (event: React.MouseEvent<unknown>, property: keyof Sale) => void;
    order: Order;
    orderBy: string;
}

export function EnhancedTableHead(props: EnhancedTableProps) {
    const { order, orderBy, onRequestSort } =
        props;
    const createSortHandler =
        (property: keyof Sale) => (event: React.MouseEvent<unknown>) => {
            onRequestSort(event, property);
        };

    return (
        <TableHead>
            <TableRow>
                {headCells.map((headCell, index) => (
                    <TableCell
                        key={headCell.id}
                        align={index === 0 ? 'left' : 'right'}
                        padding='normal'
                        sortDirection={orderBy === headCell.id ? order : false}
                    >
                        <TableSortLabel
                            active={orderBy === headCell.id}
                            direction={orderBy === headCell.id ? order : 'asc'}
                            onClick={createSortHandler(headCell.id)}
                        >
                            {headCell.label}
                            {orderBy === headCell.id ? (
                                <Box component="span" sx={visuallyHidden}>
                                    {order === 'desc' ? 'sorted descending' : 'sorted ascending'}
                                </Box>
                            ) : null}
                        </TableSortLabel>
                    </TableCell>
                ))}
            </TableRow>
        </TableHead>
    );
}