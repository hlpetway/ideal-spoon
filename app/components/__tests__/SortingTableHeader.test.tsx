import renderer from 'react-test-renderer';
import { EnhancedTableHead } from '../SortingTableHeader';

describe('EnhancedTableHead', () => {
    const onRequestSortMock = jest.fn();
    const order = 'asc';
    const orderBy = 'weekEnding';

    it('should render the table head correctly', () => {
        const component = renderer.create(
            <EnhancedTableHead onRequestSort={onRequestSortMock} order={order} orderBy={orderBy} />,
          );
          let tree = component.toJSON();
          expect(tree).toMatchSnapshot();
    });
});