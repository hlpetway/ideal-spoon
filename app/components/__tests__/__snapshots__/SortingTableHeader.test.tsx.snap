// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnhancedTableHead should render the table head correctly 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-15wwp11-MuiTableHead-root{display:table-header-group;}",
      }
    }
    data-emotion="css 15wwp11-MuiTableHead-root"
  />,
  <thead
    className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
    role={null}
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1q1u3t4-MuiTableRow-root{color:inherit;display:table-row;vertical-align:middle;outline:0;}.css-1q1u3t4-MuiTableRow-root.MuiTableRow-hover:hover{background-color:rgba(0, 0, 0, 0.04);}.css-1q1u3t4-MuiTableRow-root.Mui-selected{background-color:rgba(25, 118, 210, 0.08);}.css-1q1u3t4-MuiTableRow-root.Mui-selected:hover{background-color:rgba(25, 118, 210, 0.12);}",
        }
      }
      data-emotion="css 1q1u3t4-MuiTableRow-root"
    />
    <tr
      className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
      role={null}
    >
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-1ygcj2i-MuiTableCell-root{font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:500;font-size:0.875rem;line-height:1.5rem;letter-spacing:0.01071em;display:table-cell;vertical-align:inherit;border-bottom:1px solid rgba(224, 224, 224, 1);text-align:left;padding:16px;color:rgba(0, 0, 0, 0.87);}",
          }
        }
        data-emotion="css 1ygcj2i-MuiTableCell-root"
      />
      <th
        aria-sort="ascending"
        className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
        scope="col"
      >
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-o11grf-MuiTableSortLabel-root{cursor:pointer;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-box-pack:start;-ms-flex-pack:start;-webkit-justify-content:flex-start;justify-content:flex-start;-webkit-flex-direction:inherit;-ms-flex-direction:inherit;flex-direction:inherit;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}.css-o11grf-MuiTableSortLabel-root:focus{color:rgba(0, 0, 0, 0.6);}.css-o11grf-MuiTableSortLabel-root:hover{color:rgba(0, 0, 0, 0.6);}.css-o11grf-MuiTableSortLabel-root:hover .MuiTableSortLabel-icon{opacity:0.5;}.css-o11grf-MuiTableSortLabel-root.Mui-active{color:rgba(0, 0, 0, 0.87);}.css-o11grf-MuiTableSortLabel-root.Mui-active .MuiTableSortLabel-icon{opacity:1;color:rgba(0, 0, 0, 0.6);}",
            }
          }
          data-emotion="css o11grf-MuiTableSortLabel-root"
        />
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root{display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;position:relative;box-sizing:border-box;-webkit-tap-highlight-color:transparent;background-color:transparent;outline:0;border:0;margin:0;border-radius:0;padding:0;cursor:pointer;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;vertical-align:middle;-moz-appearance:none;-webkit-appearance:none;-webkit-text-decoration:none;text-decoration:none;color:inherit;cursor:pointer;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-box-pack:start;-ms-flex-pack:start;-webkit-justify-content:flex-start;justify-content:flex-start;-webkit-flex-direction:inherit;-ms-flex-direction:inherit;flex-direction:inherit;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}.css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root::-moz-focus-inner{border-style:none;}.css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root.Mui-disabled{pointer-events:none;cursor:default;}@media print{.css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root{-webkit-print-color-adjust:exact;color-adjust:exact;}}.css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root:focus{color:rgba(0, 0, 0, 0.6);}.css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root:hover{color:rgba(0, 0, 0, 0.6);}.css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root:hover .MuiTableSortLabel-icon{opacity:0.5;}.css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root.Mui-active{color:rgba(0, 0, 0, 0.87);}.css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root.Mui-active .MuiTableSortLabel-icon{opacity:1;color:rgba(0, 0, 0, 0.6);}",
            }
          }
          data-emotion="css 1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
        />
        <span
          className="MuiButtonBase-root MuiTableSortLabel-root Mui-active css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          WEEK ENDING
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-ty8jgw{border:0px solid;clip:rect(0 0 0 0);height:1px;margin:-1px;overflow:hidden;padding:0px;position:absolute;white-space:nowrap;width:1px;}",
              }
            }
            data-emotion="css ty8jgw"
          />
          <span
            className="MuiBox-root css-ty8jgw"
          >
            sorted ascending
          </span>
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-1hi7cb7-MuiTableSortLabel-icon{font-size:18px;margin-right:4px;margin-left:4px;opacity:0;-webkit-transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,-webkit-transform 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-transform:rotate(180deg);-moz-transform:rotate(180deg);-ms-transform:rotate(180deg);transform:rotate(180deg);}",
              }
            }
            data-emotion="css 1hi7cb7-MuiTableSortLabel-icon"
          />
          <style
            dangerouslySetInnerHTML={
              {
                "__html": ".css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon{-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;width:1em;height:1em;display:inline-block;fill:currentColor;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;-webkit-transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;font-size:1.5rem;font-size:18px;margin-right:4px;margin-left:4px;opacity:0;-webkit-transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,-webkit-transform 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;transition:opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;-webkit-transform:rotate(180deg);-moz-transform:rotate(180deg);-ms-transform:rotate(180deg);transform:rotate(180deg);}",
              }
            }
            data-emotion="css 1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
            data-testid="ArrowDownwardIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
            />
          </svg>
        </span>
      </th>
      <style
        dangerouslySetInnerHTML={
          {
            "__html": ".css-1azl6jz-MuiTableCell-root{font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:500;font-size:0.875rem;line-height:1.5rem;letter-spacing:0.01071em;display:table-cell;vertical-align:inherit;border-bottom:1px solid rgba(224, 224, 224, 1);text-align:right;padding:16px;color:rgba(0, 0, 0, 0.87);-webkit-flex-direction:row-reverse;-ms-flex-direction:row-reverse;flex-direction:row-reverse;}",
          }
        }
        data-emotion="css 1azl6jz-MuiTableCell-root"
      />
      <th
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
        scope="col"
      >
        <span
          className="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          RETAIL SALES
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
            data-testid="ArrowDownwardIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
            />
          </svg>
        </span>
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
        scope="col"
      >
        <span
          className="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          WHOLESALE SALES
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
            data-testid="ArrowDownwardIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
            />
          </svg>
        </span>
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
        scope="col"
      >
        <span
          className="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          UNITS SOLD
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
            data-testid="ArrowDownwardIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
            />
          </svg>
        </span>
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1azl6jz-MuiTableCell-root"
        scope="col"
      >
        <span
          className="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          RETAILER MARGIN
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
            data-testid="ArrowDownwardIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
            />
          </svg>
        </span>
      </th>
    </tr>
  </thead>,
]
`;
